### PR TITLE
K8SPXC-252 Auto updating system users

### PR DESCRIPTION
**Pull request by nonemax**
[SPXC-252](https://jira.percona.com/browse/SPXC-252)

### CODE REVIEW

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| scaling | failed |
| storage | failed |
| init-deploy | passed |
| recreate | passed |
| big-data | failed |
| scaling-proxysql | passed |
| limits | passed |
| self-healing | passed |
| upgrade | failed |
| restore-to-encrypted-cluster | passed |
| upgrade-consistency | passed |
| monitoring | failed |
| self-healing-advanced | passed |
| security-context | passed |
| monitoring-2-0 | passed |
| demand-backup | failed |
| operator-self-healing | passed |
| affinity | passed |
| scheduled-backup | passed |
| one-pod | passed |
| auto-tuning | failed |
| proxysql-sidecar-res-limits | passed |
| demand-backup-encrypted-with-tls | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | failed |
| storage | failed |
| big-data | failed |
| scaling | passed |
| recreate | passed |
| limits | passed |
| scaling-proxysql | passed |
| self-healing | passed |
| restore-to-encrypted-cluster | passed |
| upgrade | passed |
| monitoring | passed |
| monitoring-2-0 | passed |
| self-healing-advanced | passed |
| upgrade-consistency | passed |
| security-context | passed |
| demand-backup | passed |
| affinity | passed |
| operator-self-healing | passed |
| one-pod | failed |
| scheduled-backup | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| demand-backup-encrypted-with-tls | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| big-data | failed |
| init-deploy | failed |
| scaling | failed |
| storage | failed |
| recreate | passed |
| limits | passed |
| scaling-proxysql | failed |
| upgrade | passed |
| monitoring | failed |
| monitoring-2-0 | failed |
| self-healing | passed |
| restore-to-encrypted-cluster | failed |
| affinity | passed |
| upgrade-consistency | passed |
| security-context | failed |
| demand-backup | failed |
| self-healing-advanced | failed |
| one-pod | passed |
| operator-self-healing | passed |
| scheduled-backup | failed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| demand-backup-encrypted-with-tls | failed |

File: pkg/controller/pxc/users.go [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+const internalPrefix = "internal-"
+
+func (r *ReconcilePerconaXtraDBCluster) reconcileUsers(cr *api.PerconaXtraDBCluster) error {
```
Comment (blez) [inline pkg/controller/pxc/users.go:21]: This function will always return nil

File: pkg/controller/pxc/users.go [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+
+	err = r.client.Update(context.TODO(), &sfsPXC)
+	if err != nil {
+		return errors.Wrap(err, "update pxc sfs last-applied annotation")
+	}
+
+	return nil
+}
+
+func (r *ReconcilePerconaXtraDBCluster) restartProxy(cr *api.PerconaXtraDBCluster, newSecretDataHash string) error {
```
Comment (blez) [inline pkg/controller/pxc/users.go:180]: `newSecretDataHash` is unused

Comment (blez) [reply pkg/controller/pxc/users.go:180]: f

File: pkg/controller/pxc/users.go [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+		err = um.UpdateUsersPass(sysUsers)
+		if err != nil {
+			return errors.Wrap(err, "update sys users pass")
+		}
+	}
+
+	return nil
+}
+
+func (r *ReconcilePerconaXtraDBCluster) restartPXC(cr *api.PerconaXtraDBCluster, newSecretDataHash string, sysUsersSecretObj *corev1.Secret) error {
```
Comment (blez) [inline pkg/controller/pxc/users.go:157]: `sysUsersSecretObj` is unused

File: pkg/controller/pxc/users.go (hunk 2 above)
Comment (blez) [inline pkg/controller/pxc/users.go:180]: `newSecretDataHash` is unused

File: pkg/controller/pxc/users.go [hunk 4]
```diff
@@ -0,0 +1,284 @@
...
+		err = r.restartPXC(cr, newSecretDataHash, &sysUsersSecretObj)
+		if err != nil {
+			return errors.Wrap(err, "restart pxc")
+		}
+	}
+
+	return nil
+}
+
+func (r *ReconcilePerconaXtraDBCluster) manageSysUsers(cr *api.PerconaXtraDBCluster, sysUsersSecretObj, internalSysSecretObj *corev1.Secret, restartPXC, restartProxy *bool) error {
```
Comment (blez) [inline pkg/controller/pxc/users.go:99]: I think `restartPXC`, `restartProxy` should not be parameters, but returned result.

File: pkg/controller/pxc/users.go [hunk 5]
```diff
@@ -0,0 +1,284 @@
...
+		sfsPXC.Annotations = make(map[string]string)
+	}
+	sfsPXC.Spec.Template.Annotations["last-applied-secret"] = newSecretDataHash
+
+	err = r.client.Update(context.TODO(), &sfsPXC)
+	if err != nil {
+		return errors.Wrap(err, "update pxc sfs last-applied annotation")
+	}
+
+	return nil
```
Comment (blez) [inline pkg/controller/pxc/users.go:177]: It is not mandatory, but can be replaced by
`return errors.Wrap(err, "update pxc sfs last-applied annotation")`

File: pkg/pxc/users/users.go [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+
+type SysUser struct {
+	Name  string   `yaml:"username"`
+	Pass  string   `yaml:"password"`
+	Hosts []string `yaml:"hosts"`
+}
+
+func NewManager(hosts []string, rootPass string) (Manager, error) {
+	var um Manager
+	var err error
```
Comment (blez) [inline pkg/pxc/users/users.go:22]: this error will be always nil

File: pkg/pxc/users/users.go [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+	db *sql.DB
+}
+
+type SysUser struct {
+	Name  string   `yaml:"username"`
+	Pass  string   `yaml:"password"`
+	Hosts []string `yaml:"hosts"`
+}
+
+func NewManager(hosts []string, rootPass string) (Manager, error) {
```
Comment (blez) [inline pkg/pxc/users/users.go:20]: Why are the multiple hosts allowed? As I see just service name used to connect to PXC.

File: pkg/pxc/users/users.go [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+func (u *Manager) UpdateUsersPass(users []SysUser) error {
+	defer u.db.Close()
+	tx, err := u.db.Begin()
+	if err != nil {
+		return errors.Wrap(err, "begin transaction")
+	}
+
+	_, err = tx.Exec("FLUSH PRIVILEGES")
+	if err != nil {
+		tx.Rollback()
```
Comment (blez) [inline pkg/pxc/users/users.go:47]: I know it's strange, but just for me :)
If you ignore the returned errors could you please do it explicit way?
`_ = tx.Rollback()`
My linter goes crazy without `_ =`

File: pkg/controller/pxc/users.go [hunk 9]
```diff
@@ -0,0 +1,284 @@
...
+	err = r.client.Get(context.TODO(),
+		types.NamespacedName{
+			Namespace: cr.Namespace,
+			Name:      secretName,
+		},
+		&internalSysUsersSecretObj,
+	)
+	if err != nil && !k8serrors.IsNotFound(err) {
+		return internalSysUsersSecretObj, errors.Wrap(err, "get internal sys users secret")
+	} else if k8serrors.IsNotFound(err) {
```
Comment (blez) [inline pkg/controller/pxc/users.go:228]: else is not needed

Comment (nonemax) [reply pkg/controller/pxc/users.go]: fixed https://github.com/percona/percona-xtradb-cluster-operator/pull/418/commits/9a770955143b4e0e1b00e53653fdb45ab333357e

File: pkg/pxc/users/users.go (hunk 8 above)
Comment (nonemax) [reply pkg/pxc/users/users.go]: https://github.com/percona/percona-xtradb-cluster-operator/pull/418/commits/9a770955143b4e0e1b00e53653fdb45ab333357e

File: pkg/pxc/users/users.go (hunk 7 above)
Comment (nonemax) [reply pkg/pxc/users/users.go]: removed https://github.com/percona/percona-xtradb-cluster-operator/pull/418/commits/9a770955143b4e0e1b00e53653fdb45ab333357e

File: pkg/pxc/users/users.go (hunk 6 above)
Comment (nonemax) [reply pkg/pxc/users/users.go]: https://github.com/percona/percona-xtradb-cluster-operator/pull/418/commits/9a770955143b4e0e1b00e53653fdb45ab333357e

File: pkg/controller/pxc/users.go (hunk 5 above)
Comment (nonemax) [reply pkg/controller/pxc/users.go:171]: I think it can not

File: pkg/controller/pxc/users.go (hunk 4 above)
Comment (nonemax) [reply pkg/controller/pxc/users.go]: done https://github.com/percona/percona-xtradb-cluster-operator/pull/418/commits/9a770955143b4e0e1b00e53653fdb45ab333357e

File: pkg/controller/pxc/users.go (hunk 2 above)
Comment (nonemax) [reply pkg/controller/pxc/users.go]: removed https://github.com/percona/percona-xtradb-cluster-operator/pull/418/commits/9a770955143b4e0e1b00e53653fdb45ab333357e

File: pkg/controller/pxc/users.go (hunk 3 above)
Comment (nonemax) [reply pkg/controller/pxc/users.go]: removed https://github.com/percona/percona-xtradb-cluster-operator/pull/418/commits/9a770955143b4e0e1b00e53653fdb45ab333357e

File: pkg/controller/pxc/users.go (hunk 1 above)
Comment (nonemax) [reply pkg/controller/pxc/users.go:21]: fixed https://github.com/percona/percona-xtradb-cluster-operator/pull/418/commits/9a770955143b4e0e1b00e53653fdb45ab333357e

File: pkg/pxc/users/users.go [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
+	}
+
+	_, err = tx.Exec("FLUSH PRIVILEGES")
+	if err != nil {
+		errT := tx.Rollback()
+		if errT != nil {
+			return errors.Errorf("flush privileges: %v, tx rollback: %v", err, errT)
+		}
+		return errors.Wrap(err, "flush privileges")
+	}
```
Comment (delgod) [inline pkg/pxc/users/users.go:46]: why do we need to flush privileges two times in one transaction? I think only the second run in the transaction has a real effect.

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| scaling | passed |
| storage | failed |
| big-data | passed |
| init-deploy | passed |
| recreate | passed |
| scaling-proxysql | passed |
| limits | passed |
| monitoring | passed |
| self-healing | passed |
| upgrade | passed |
| restore-to-encrypted-cluster | passed |
| monitoring-2-0 | passed |
| self-healing-advanced | passed |
| upgrade-consistency | passed |
| security-context | passed |
| operator-self-healing | passed |
| affinity | passed |
| demand-backup | passed |
| one-pod | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| scheduled-backup | passed |
| demand-backup-encrypted-with-tls | passed |

File: Gopkg.lock [hunk 11]
```diff
@@ -789,6 +789,7 @@
     "github.com/operator-framework/operator-sdk/pkg/ready",
     "github.com/operator-framework/operator-sdk/version",
     "github.com/pkg/errors",
+    "gopkg.in/yaml.v2",
```
Comment (delgod) [inline Gopkg.lock:792]: why it is needed?

File: pkg/controller/pxc/users.go [hunk 12]
```diff
@@ -0,0 +1,280 @@
...
+
+func (r *ReconcilePerconaXtraDBCluster) reconcileUsers(cr *api.PerconaXtraDBCluster) error {
+	if cr.Status.Status != api.AppStateReady {
+		return nil
+	}
+
+	return r.handleSysUsersSecret(cr)
+}
+
+func (r *ReconcilePerconaXtraDBCluster) handleSysUsersSecret(cr *api.PerconaXtraDBCluster) error {
```
Comment (delgod) [inline pkg/controller/pxc/users.go:29]: do we really need to have `handleSysUsersSecret` as a separate function?
```suggestion
```

File: pkg/controller/pxc/users.go [hunk 13]
```diff
@@ -0,0 +1,280 @@
...
+			continue
+		}
+
+		switch name {
+		case "root":
+			restartProxy = true
+			hosts = []string{"localhost", "%"}
+		case "xtrabackup":
+			restartProxy = true
+			restartPXC = true
```
Comment (delgod) [inline pkg/controller/pxc/users.go:113]: I know that we pass xtrabackup user to ProxySQL, but I don't see where we use it 😞 
```suggestion
		case "xtrabackup":
			restartPXC = true
```

File: pkg/controller/pxc/users.go [hunk 14]
```diff
@@ -0,0 +1,280 @@
...
+			hosts = []string{"localhost"}
+		case "monitor":
+			restartProxy = true
+			hosts = []string{"10.%", "%"}
```
Comment (delgod) [inline pkg/controller/pxc/users.go:117]: also needed to add PXC if PMM is enabled
```suggestion
                      if cr.Spec.PMM.Enabled {
                          restartPXC = true
                      }
			hosts = []string{"10.%", "%"}
```

File: pkg/controller/pxc/users.go [hunk 15]
```diff
@@ -0,0 +1,280 @@
...
+		case "clustercheck":
+			restartProxy = true
+			restartPXC = true
```
Comment (delgod) [inline pkg/controller/pxc/users.go:120]: not sure about ProxySQL, is it used?
```suggestion
		case "clustercheck":
			restartPXC = true
```

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | failed |
| big-data | passed |
| scaling | passed |
| recreate | passed |
| init-deploy | passed |
| scaling-proxysql | passed |
| limits | passed |
| self-healing | passed |
| restore-to-encrypted-cluster | passed |
| upgrade | passed |
| monitoring | passed |
| self-healing-advanced | passed |
| upgrade-consistency | passed |
| monitoring-2-0 | passed |
| demand-backup | passed |
| security-context | passed |
| operator-self-healing | passed |
| affinity | passed |
| one-pod | passed |
| auto-tuning | passed |
| scheduled-backup | failed |
| proxysql-sidecar-res-limits | passed |
| demand-backup-encrypted-with-tls | failed |

File: pkg/controller/pxc/users.go [hunk 16]
```diff
@@ -0,0 +1,280 @@
...
+	pvcProxy := corev1.PersistentVolumeClaim{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: cr.Namespace,
+			Name:      "proxydata-" + cr.Name + "-proxysql-0",
+		},
+	}
+	err := r.client.Delete(context.TODO(), &pvcProxy)
+	if err != nil {
+		return errors.Wrap(err, "delete proxy pvc")
+	}
```
Comment (delgod) [inline pkg/controller/pxc/users.go:184]: wow, why we need to delete PVC?

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| scaling | passed |
| big-data | passed |
| init-deploy | passed |
| recreate | passed |
| storage | failed |
| scaling-proxysql | passed |
| limits | passed |
| monitoring | passed |
| restore-to-encrypted-cluster | passed |
| self-healing | passed |
| upgrade | passed |
| monitoring-2-0 | passed |
| affinity | passed |
| upgrade-consistency | passed |
| demand-backup | passed |
| self-healing-advanced | failed |
| security-context | passed |
| one-pod | passed |
| operator-self-healing | passed |
| auto-tuning | failed |
| proxysql-sidecar-res-limits | passed |
| scheduled-backup | failed |
| demand-backup-encrypted-with-tls | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | failed |
| recreate | passed |
| big-data | passed |
| scaling | passed |
| init-deploy | failed |
| scaling-proxysql | passed |
| self-healing | passed |
| restore-to-encrypted-cluster | passed |
| upgrade | passed |
| limits | passed |
| monitoring | passed |
| self-healing-advanced | passed |
| upgrade-consistency | passed |
| demand-backup | failed |
| security-context | passed |
| operator-self-healing | passed |
| monitoring-2-0 | passed |
| scheduled-backup | passed |
| affinity | passed |
| one-pod | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| demand-backup-encrypted-with-tls | passed |

File: pkg/controller/pxc/users.go (hunk 16 above)
Comment (nonemax) [reply pkg/controller/pxc/users.go]: removed https://github.com/percona/percona-xtradb-cluster-operator/pull/418/commits/6c99594a7fcf4bdb02009113fe96749e10a465f9

File: pkg/controller/pxc/users.go (hunk 15 above)
Comment (nonemax) [reply pkg/controller/pxc/users.go]: https://github.com/percona/percona-xtradb-cluster-operator/pull/418/commits/6c99594a7fcf4bdb02009113fe96749e10a465f9

File: pkg/controller/pxc/users.go (hunk 14 above)
Comment (nonemax) [reply pkg/controller/pxc/users.go]: https://github.com/percona/percona-xtradb-cluster-operator/pull/418/commits/6c99594a7fcf4bdb02009113fe96749e10a465f9

File: pkg/controller/pxc/users.go (hunk 13 above)
Comment (nonemax) [reply pkg/controller/pxc/users.go]: https://github.com/percona/percona-xtradb-cluster-operator/pull/418/commits/6c99594a7fcf4bdb02009113fe96749e10a465f9

File: pkg/pxc/users/users.go (hunk 10 above)
Comment (nonemax) [reply pkg/pxc/users/users.go]: removed https://github.com/percona/percona-xtradb-cluster-operator/pull/418/commits/3c3d08534e84f25c5701e9a930f3dc3783fcac9e

File: Gopkg.lock (hunk 11 above)
Comment (nonemax) [reply Gopkg.lock]: removed https://github.com/percona/percona-xtradb-cluster-operator/pull/418/commits/c38f1fc0f97c48e615fca431ae5db41561f42adf

File: pkg/controller/pxc/users.go [hunk 17]
```diff
@@ -0,0 +1,331 @@
...
+	for name, pass := range sysUsersSecretObj.Data {
+		hosts := []string{}
+
+		if string(sysUsersSecretObj.Data[name]) == string(internalSysSecretObj.Data[name]) {
+			continue
+		}
+
+		switch name {
+		case "root":
+			syncUsers = true
```
Comment (delgod) [inline pkg/controller/pxc/users.go:107]:
```suggestion
			syncProxySQLUsers = true
```

File: pkg/controller/pxc/users.go [hunk 18]
```diff
@@ -0,0 +1,331 @@
...
+	}
+
+	if len(sysUsers) > 0 {
+		err = um.UpdateUsersPass(sysUsers)
+		if err != nil {
+			return restartPXC, restartProxy, errors.Wrap(err, "update sys users pass")
+		}
+	}
+
+	if updProxyUsers && len(proxyUsers) > 0 {
```
Comment (delgod) [inline pkg/controller/pxc/users.go:153]: `updProxyUsers` not needed
```suggestion
	if len(proxyUsers) > 0 {
```

File: pkg/controller/pxc/users.go [hunk 19]
```diff
@@ -0,0 +1,331 @@
...
+			restartPXC = true
+			hosts = []string{"localhost"}
+		case "monitor":
+			restartProxy = true
+			updProxyUsers = true
+			proxyUsers = append(proxyUsers, users.SysUser{Name: name, Pass: string(pass)})
+
+			if cr.Spec.PMM.Enabled {
+				restartPXC = true
+			}
```
Comment (delgod) [inline pkg/controller/pxc/users.go:119]: it is needed to update also password which used by pmm-client
```suggestion
			syncProxySQLUsers = true
			proxyUsers = append(proxyUsers, users.SysUser{Name: name, Pass: string(pass)})

			if cr.Spec.PMM.Enabled {
				restartPXC = true
				restartProxy = true
			}
```

File: pkg/controller/pxc/users.go [hunk 20]
```diff
@@ -0,0 +1,331 @@
...
+			return restartPXC, restartProxy, errors.Wrap(err, "sync users")
+		}
+	}
+
+	return restartPXC, restartProxy, nil
+}
+
+func (r *ReconcilePerconaXtraDBCluster) syncPXCUsersWithProxySQL(cr *api.PerconaXtraDBCluster) error {
+	// sync users if ProxySql enabled
+	if cr.Spec.ProxySQL != nil && cr.Spec.ProxySQL.Size > 0 {
```
Comment (delgod) [inline pkg/controller/pxc/users.go:172]: should we also check cr.Spec.ProxySQL.Enabled ?
```suggestion
	if cr.Spec.ProxySQL != nil && cr.Spec.ProxySQL.Enabled && cr.Spec.ProxySQL.Size > 0 {
```

File: pkg/controller/pxc/users.go [hunk 21]
```diff
@@ -0,0 +1,331 @@
...
+	if err != nil {
+		return errors.Wrap(err, "manage sys users")
+	}
+
+	err = r.updateInternalSysUsersSecret(cr, &sysUsersSecretObj)
+	if err != nil {
+		return errors.Wrap(err, "update internal sys users secret")
+	}
+
+	if restartProxy && cr.Spec.ProxySQL != nil && cr.Spec.ProxySQL.Size > 0 {
```
Comment (delgod) [inline pkg/controller/pxc/users.go:71]: 1) should we also check `cr.Spec.ProxySQL.Enabled`?
2) we just update hash, is it really needed to check size?
```suggestion
	if restartProxy && cr.Spec.ProxySQL != nil && cr.Spec.ProxySQL.Enabled {
```

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | failed |
| big-data | failed |
| scaling | passed |
| recreate | passed |
| storage | failed |
| scaling-proxysql | passed |
| restore-to-encrypted-cluster | failed |
| limits | passed |
| self-healing | passed |
| upgrade | passed |
| monitoring | passed |
| upgrade-consistency | passed |
| demand-backup | failed |
| security-context | passed |
| self-healing-advanced | passed |
| monitoring-2-0 | passed |
| operator-self-healing | passed |
| affinity | passed |
| scheduled-backup | failed |
| one-pod | passed |
| auto-tuning | passed |
| demand-backup-encrypted-with-tls | failed |
| proxysql-sidecar-res-limits | passed |

File: pkg/controller/pxc/users.go (hunk 19 above)
Comment (delgod) [reply pkg/controller/pxc/users.go]: sorry my mistake

File: pkg/controller/pxc/users.go [hunk 22]
```diff
@@ -0,0 +1,331 @@
...
+	"k8s.io/apimachinery/pkg/types"
+)
+
+const internalPrefix = "internal-"
+
+func (r *ReconcilePerconaXtraDBCluster) reconcileUsers(cr *api.PerconaXtraDBCluster) error {
+	if cr.Status.Status != api.AppStateReady {
+		return nil
+	}
+
```
Comment (delgod) [inline pkg/controller/pxc/users.go:26]: check CR version

Comment (nonemax) [reply pkg/controller/pxc/users.go:26]: done https://github.com/percona/percona-xtradb-cluster-operator/pull/418/commits/b26d5047e5d7ab5ca27cda071d72279d01ba49fa

File: pkg/controller/pxc/users.go (hunk 21 above)
Comment (nonemax) [reply pkg/controller/pxc/users.go]: https://github.com/percona/percona-xtradb-cluster-operator/pull/418/commits/b26d5047e5d7ab5ca27cda071d72279d01ba49fa

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| scaling | passed |
| recreate | passed |
| big-data | passed |
| init-deploy | failed |
| storage | passed |
| self-healing | passed |
| scaling-proxysql | passed |
| restore-to-encrypted-cluster | passed |
| limits | passed |
| upgrade | passed |
| self-healing-advanced | failed |
| upgrade-consistency | passed |
| monitoring | passed |
| operator-self-healing | passed |
| security-context | passed |
| demand-backup | failed |
| monitoring-2-0 | passed |
| affinity | passed |
| one-pod | passed |
| scheduled-backup | failed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| demand-backup-encrypted-with-tls | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| recreate | passed |
| restore-to-encrypted-cluster | passed |
| demand-backup | failed |
| scaling | passed |
| scaling-proxysql | passed |
| upgrade | passed |
| upgrade-consistency | passed |
| security-context | passed |
| storage | passed |
| self-healing | passed |
| self-healing-advanced | passed |
| init-deploy | failed |
| big-data | passed |
| limits | passed |
| monitoring | passed |
| monitoring-2-0 | passed |
| affinity | passed |
| one-pod | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| operator-self-healing | passed |
| scheduled-backup | failed |
| demand-backup-encrypted-with-tls | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| big-data | passed |
| recreate | passed |
| restore-to-encrypted-cluster | passed |
| demand-backup | failed |
| scaling | passed |
| scaling-proxysql | passed |
| upgrade | passed |
| upgrade-consistency | passed |
| security-context | passed |
| init-deploy | failed |
| storage | passed |
| self-healing | passed |
| self-healing-advanced | passed |
| operator-self-healing | passed |
| scheduled-backup | failed |
| limits | passed |
| monitoring | passed |
| monitoring-2-0 | passed |
| affinity | passed |
| one-pod | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| demand-backup-encrypted-with-tls | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| recreate | passed |
| restore-to-encrypted-cluster | passed |
| demand-backup | failed |
| init-deploy | passed |
| scaling | passed |
| scaling-proxysql | passed |
| upgrade | passed |
| upgrade-consistency | passed |
| security-context | passed |
| storage | passed |
| self-healing | passed |
| self-healing-advanced | passed |
| operator-self-healing | passed |
| big-data | passed |
| limits | passed |
| monitoring | passed |
| monitoring-2-0 | passed |
| affinity | passed |
| one-pod | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| scheduled-backup | failed |
| demand-backup-encrypted-with-tls | passed |

**Pull request merged**